microsoft/testsuites/network/sriov.py

```diff
@@ -310,9 +320,264 @@ def sriov_add_max_nics_validation(
+        for node in environment.nodes.list():
+            self._load_module(node, module_in_used)
+        vm_nics = self._initialize_nic_info(environment)
+        self._sriov_vf_connection_test(environment, vm_nics)
+
+    @TestCaseMetadata(
+        description="""
+        This case verify below two kernel patches.
+        1. hv_netvsc: Sync offloading features to VF NIC
+           https://github.com/torvalds/linux/commit/68622d071e555e1528f3e7807f30f73311c1acae#diff-007213ba7199932efdb096be47d209a2f83e4d425c486b3adaba861d0a0c80c5 # noqa: E501
+        2. hv_netvsc: Allow scatter-gather feature to be tunable
+           https://github.com/torvalds/linux/commit/b441f79532ec13dc82d05c55badc4da1f62a6141#diff-007213ba7199932efdb096be47d209a2f83e4d425c486b3adaba861d0a0c80c5 # noqa: E501
+
+        Steps,
+        1. Change scatter-gather feature on synthetic nic,
+         verify the the feature status sync to the VF dynamically.
+        2. Disable and enable sriov,
+         check the scatter-gather feature status keep consistent in VF.
+        """,
+        priority=2,
+        requirement=simple_requirement(
+            min_count=2,
+            network_interface=schema.NetworkInterfaceOptionSettings(
+                nic_count=3,
+                data_path=schema.NetworkDataPath.Sriov,
+            ),
+        ),
+    )
+    def sriov_ethtool_offload_setting(self, environment: Environment) -> None:
+        server_node = cast(RemoteNode, environment.nodes[0])
+        client_node = cast(RemoteNode, environment.nodes[1])
+        client_ethtool = client_node.tools[Ethtool]
+        vm_nics = self._initialize_nic_info(environment)
+        # skip test if scatter-gather can't be updated
+        for _, client_nic_info in vm_nics[client_node.name].items():
+            device_sg_settings = client_ethtool.get_device_sg_settings(
+                client_nic_info.upper, True
+            )
+            if device_sg_settings.sg_fixed:
+                raise SkippedException(
+                    "scatter-gather is fixed can cannot be changed for device"
+                    f" {client_nic_info.upper}. Skipping test."
+                )
+            else:
+                break
+        # save original enabled features
+        device_enabled_features_origin = client_ethtool.get_all_device_enabled_features(
+            True
+        )
+
+        # preparation work before launch iperf3
+        self._stop_firewall(environment)
+
+        # run iperf3 on server side and client side
+        # iperfResults.log stored client side log
+        source_iperf3 = server_node.tools[Iperf3]
+        dest_iperf3 = client_node.tools[Iperf3]
+        source_iperf3.run_as_server_mode()
+        dest_iperf3.run_as_client_mode(
+            1800, server_node.internal_address, "iperfResults.log", True
+        )
+
+        # wait for a while then check any error shown up in iperfResults.log
+        sleep(10)
```

Comment:
Do not put sleep here. change the `run_as_client` to `run_as_client` and `run_as_client_async`. Check other  sync/async pattern to implement it. Check if client process exited or not to detect error.